app/controllers/vm_cloud_controller.rb

```diff
@@ -113,16 +113,12 @@ def live_migrate
   def live_migrate_form_fields
     assert_privileges("instance_live_migrate")
     @record = find_by_id_filtered(VmOrTemplate, params[:id])
-    clusters = []
-    hosts = []
-    @record.ext_management_system.find_filtered_children("ems_clusters").each do |c|
-      clusters << {:id => c.id, :name => c.name}
+    clusters = @record.ext_management_system.ems_clusters.map do |c|
+      {:id => c.id, :name => c.name}
     end
-    @record.ext_management_system.find_filtered_children("hosts").each do |h|
-      hosts << {:id => h.id, :name => h.name, :cluster_id => h.emd_cluster.id}
+    hosts = @record.ext_management_system.hosts.map do |h|
+      {:id => h.id, :name => h.name, :cluster_id => h.emd_cluster.id}
     end
-    clusters.sort
-    hosts.sort
```

Comment:
@mansam this `sort` did nothing :smile: If you want to sort the original value you'd have to use `sort!` or assing it, like `hosts = hosts.sort`
And then it would sort the contents of the array, which are Hashes. And they dont sort really good in ruby2.2 (Hashes will be sorted by being subset of another in 2.3)
I think what you wanted is e.g. sorting by name? 
But you could do that here already `@record.ext_management_stystem.hosts.order(:name)`